@xen-orchestra/fs/src/fs.spec.js

```diff
@@ -0,0 +1,132 @@
+/* eslint-env jest */
+
+import { getHandler } from '.'
+import { tmpdir } from 'os'
+import getStream from 'get-stream'
+
+// https://gist.github.com/julien-f/3228c3f34fdac01ade09
+const unsecureRandomBytes = n => {
+  const bytes = Buffer.alloc(n)
+
+  const odd = n & 1
+  for (let i = 0, m = n - odd; i < m; i += 2) {
+    bytes.writeUInt16BE((Math.random() * 65536) | 0, i)
+  }
+
+  if (odd) {
+    bytes.writeUInt8((Math.random() * 256) | 0, n - 1)
+  }
+
+  return bytes
+}
+
+const TEST_DATA = unsecureRandomBytes(1024)
+
+;[`file://${tmpdir()}`].forEach(url => {
+  describe(url, () => {
+    let handler
+    const testDir = `xo-fs-tests-${Date.now()}`
+    const testFile = `${testDir}/file`
+
+    beforeAll(() => {
+      handler = getHandler({ url })
+    })
+    afterAll(() => {
+      handler = undefined
+    })
+
+    afterEach(async () => {
+      await handler.rmdir(testDir, { recursive: true }).catch(error => {
+        if (error.code !== 'ENOENT') {
+          throw error
+        }
+      })
+    })
+
+    describe('#test()', () => {
+      it('tests the remote appears to be working', async () => {
+        expect(await handler.test()).toMatchSnapshot()
+      })
+    })
+
+    describe('outputFile', () => {
+      it('writes data to a file', async () => {
+        expect(await handler.outputFile(testFile, TEST_DATA)).toEqual(
```

Comment:
What are you testing here?
What's `outputFiles`'s return value?

Reply:
Simply `await outputFile` and yes, check the file contain what you expect with `readFile`.